### PR TITLE
Remove qesap_terraform_clean_up_retry

### PR DESCRIPTION
**Pull request by mpagot**
Method qesap_terraform_clean_up_retry is only called once, and only from another method within the same library: move the code directly in the caller. Add more UT coverage for the caller.

Follow up of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18159

- Related ticket: TEAM-6903

# Verification run:

### sle-15-SP6-HanaSr-Azure-Byos-x86_64-Build15-SP6_2024-09-30T06:43:12Z-hanasr_azure_test_saptune_spn az_Standard_E4s_v3
 -> http://openqaworker15.qa.suse.cz/tests/298767 :green_circle: 
 
 ### sle-15-SP6-HanaSr-Aws-Byos-x86_64-Build15-SP6_2024-09-30T06:43:12Z-hanasr_aws_test_saptune_fencing_native_crash ec2_r4.8xlarge
 -> http://openqaworker15.qa.suse.cz/tests/298768 :green_circle: failure is later after the deployment, so not influenced by code in this PR

### sle-15-SP6-Qesap-Azure-Payg-x86_64-BuildLATEST_AZURE_SLE15_6_PAYG-qesap_azure_saptune_test
-> http://openqaworker15.qa.suse.cz/tests/298769 :green_circle: 

### sle-15-SP4-Qesap-Gcp-Payg-x86_64-BuildLATEST_GCE_SLE15_4_PAYG-qesap_gcp_sapconf_test
 -> http://openqaworker15.qa.suse.cz/tests/298770  :green_circle: failure is later after the deployment, so not influenced by code in this PR

### CODE REVIEW

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: lib/qesapdeployment.pm [hunk 1]
```diff
@@ -595,7 +592,6 @@ sub qesap_execute_conditional_retry {
     foreach (qw(cmd logname error_string)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
 
     my $verbose = $args{verbose} ? "--verbose" : "";
-    $args{cmd_options} //= '';
```
Comment (mpagot) [inline lib/qesapdeployment.pm:598]: it was a never used argument within the method implementation of in any method call. So I removed it

**Review by lpalovsky — APPROVED** [review]
LGTM

**Review by lilyeyes — COMMENTED** [review]
LGTM.

**Pull request merged**